src/makeInfinityAdder.js

```diff
@@ -6,7 +6,28 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+  let cleanResult = 0;
+
+  const adder = (x) => {
+    if (x >= 0) {
+      result += x;
+    }
+
+    if (x === undefined && result > 0) {
+      cleanResult++;
+
+      return cleanResult === 2 ? 0 : result;
+    }
+
+    if (x === undefined && result === 0) {
+      return 0;
+    }
```

Comment:
```suggestion
    if (x === undefined) {
      const result = x;
      
      x = 0;
      
      return result;
    }
```

in this case cleanResult is redundand

Reply:
я не розумію, як такий код буде працювать?
мені ж треба рахувать, щоб при adder() перший раз виводило суму, а при adder() 2 раз, збивало на 0.
як це реалізувать?

якщо     if (x === undefined) {
      const result = x;
      
      x = 0;
      
      return result;}
      
      то функція при першому ж виклику буде повертать 0, хіба ні?